### ISSUE TITLE
Exception on Room.disconnect()

### ISSUE THREAD

**Issue by JoaoMosmann** [gamestdio/colyseus#50]
I'm getting this exception when I do a `room.disconnect();`. Any idea what it could be?
```
  /app/node_modules/colyseus/lib/Utils.js:12 
      arr.length = len; 
                 ^ 
  RangeError: Invalid array length 
      at Object.spliceOne (/app/node_modules/colyseus/lib/Utils.js:12:16) 
      at MatchRoom.Room._onLeave (/app/node_modules/colyseus/lib/Room.js:145:17) 
      at /app/node_modules/colyseus/lib/Server.js:145:71 
      at Array.forEach (native) 
      at Server.onDisconnect (/app/node_modules/colyseus/lib/Server.js:145:33) 
      at process.nextTick (/app/node_modules/uws/uws.js:440:27) 
      at _combinedTickCallback (internal/process/next_tick.js:73:7) 
      at process._tickCallback (internal/process/next_tick.js:104:9)  
```

**Comment by AnubisCode** [comment]
@JoaoMosmann I got same error and fix by adding:
```
if(this.clients.indexOf(client)!=-1) {
   Utils_1.spliceOne(this.clients, this.clients.indexOf(client));
}
```
to Room.js class

**Comment by endel** [comment]
Oops, looks like there are no tests for `Room.disconnect()` 😅

**Comment by endel** [comment]
It's now fixed on `v0.4.17`. Thanks for reporting @JoaoMosmann ❤️
